packages/tupaia-web/src/api/queries/useEntities.ts

```diff
@@ -7,21 +7,25 @@ import { useQuery, QueryObserverOptions } from 'react-query';
 import { EntityResponse } from '../../types';
 import { get } from '../api';
 
+type EntitiesResponse = EntityResponse[];
+
 export const useEntities = (
   projectCode?: string,
   entityCode?: string,
   axiosConfig?: AxiosRequestConfig,
   queryOptions?: QueryObserverOptions,
 ) => {
-  const enabled =
-    queryOptions?.enabled === undefined ? !!projectCode && !!entityCode : queryOptions.enabled;
+  let enabled = !!projectCode && !!entityCode;
+
+  if (queryOptions?.enabled !== undefined) {
+    enabled = enabled && queryOptions.enabled;
+  }
 
   return useQuery(
-    ['entities', projectCode, entityCode, axiosConfig],
-    (): Promise<EntityResponse[]> =>
-      get(`entities/${projectCode}/${entityCode}`, {
+    ['entities', projectCode, entityCode, axiosConfig, queryOptions],
+    async (): Promise<EntitiesResponse> => {
+      return get(`entities/${projectCode}/${entityCode}`, {
```

Comment:
Quick question - is there a particular reason for adding `async` and `return` here?

Reply:
just debugging